src/main/java/dev/streamx/sling/connector/paths/PathsExtractionConfig.java

```diff
@@ -0,0 +1,106 @@
+ * <ul>
+ *   <li>Determine how to search for and extract paths via regex patterns.</li>
+ *   <li>Filter out unwanted paths.</li>
+ *   <li>Validate whether a {@link Resource} meets certain criteria based on path
+ *       and JCR primary node type before being recognized.</li>
+ * </ul>
+ *
+ * <p>The textual representation of a {@link Resource} is obtained by issuing an internal Sling request.
+ * If {@link #postfix$_$to$_$append()} is provided, that postfix is appended to the {@link Resource}
+ * path before the request. Then the specified regexes are applied to the response to discover
+ * candidate paths, which can subsequently be excluded or accepted based on the provided
+ * filtering regexes.
+ */
+@ObjectClassDefinition
+public @interface PathsExtractionConfig {
+
+  /**
+   * An array of regex patterns used to search for paths within the textual representation of a
+   * {@link Resource}. Each pattern must capture the path in the <em>first capturing group</em>
+   * (i.e., {@code group(1)}).
+   *
+   * @return array of regex patterns used to search for paths within the textual representation of a
+   * {@link Resource}
+   */
+  @AttributeDefinition(
+      name = "Search Regexes",
+      description = "Array of regex patterns used to search for paths within "
+          + "the textual representation of a Resource.",
+      type = AttributeType.STRING
+  )
+  String[] search_regexes() default {};
+
+  /**
+   * A regex pattern used to exclude certain paths from the final extracted results. Any path
+   * matching this pattern will be omitted.
+   *
+   * @return regex pattern used to exclude certain paths from the final extracted results.
+   */
+  @AttributeDefinition(
+      name = "Regex to Exclude from Result",
+      description = "Regex pattern used to exclude certain paths from the final extracted results.",
+      type = AttributeType.STRING,
+      defaultValue = StringUtils.EMPTY
+  )
+  String exclude$_$from$_$result_regex() default StringUtils.EMPTY;
+
+  /**
+   * Specifies a postfix to append to the {@link Resource} path before performing the internal Sling
+   * request to retrieve the text representation of that {@link Resource}. This can help ensure the
+   * {@link Resource} responds with the desired format. For example, appending {@code .html} might
+   * trigger a specific rendering script.
+   *
+   * @return postfix to append to the {@link Resource} path before performing the internal Sling
+   * request to retrieve the text representation of that {@link Resource}.
+   */
+  @AttributeDefinition(
+      name = "Postfix to Append",
+      description = "Postfix to append to the Resource path before performing the internal Sling request to retrieve the text representation of that Resource. Specified value must not be prepended with a dot (.).",
+      type = AttributeType.STRING,
+      defaultValue = StringUtils.EMPTY
+  )
+  String postfix$_$to$_$append() default StringUtils.EMPTY;
```

Comment:
we discussed before that these 3 props related to the resource passed to the RelatedResourcesSelector.getRelatedResources should be marked in name. Maybe resource word at the begining and in the names of the :
Postfix to Append -> Resource Path Postfix to Append
postfix$_$to$_$append() -> resource$_$path$_$postfix$_$to$_$append()

Path Regex -> Resource Path Regex
required$_$path_regex -> required$_resource$_$path_regex

Node Type Regex -> Resource Node Primary Type Regex
required$_$resource&_$primary$_$node$_$type_regex

Reply:
and for reference related props:
Search Regexes -> References Search Regexes 
search_regexes -> references_search_regexes

Regex to Exclude from Result -> References Regex to Exclude from Result
exclude$_$from$_$result_regex -> references$_$exclude$_$from$_$result_regex